ext/CatalystHomotopyContinuationExtension/homotopy_continuation_extension.jl

```diff
@@ -104,4 +105,13 @@ function filter_negative_f(sols; neg_thres=-1e-20)
         (neg_thres < sol[idx] < 0) && (sol[idx] = 0)
     end
     return filter(sol -> all(>=(0), sol), sols)
+end
+
+# Sometimes (when polynomials are created from hybrid CRN/DAEs), the steady state polynomial have the wrong type.
+# This converts it to the correct type, which homotopy continuation can handle.
+const WRONG_POLY_TYPE = Vector{DynamicPolynomials.Polynomial{DynamicPolynomials.Commutative{DynamicPolynomials.CreationOrder}, DynamicPolynomials.Graded{DynamicPolynomials.LexOrder}}}
```

Comment:
Sometimes you get a polynomial system of `WRONG_POLY_TYPE`, which can be converted to `CORRECT_POLY_TYPE`. Long-term this could probably be sorted out in HC, but until then I put this bit in here.